### PR TITLE
Migrate ask order from Bucket to Map

### PR DESCRIPTION
**Pull request by brashidnia-figure**
Migrate the ask order storage from `cosmwasm-storage::Bucket` to `cw-storage-plus:Map`
- Update the src, and tests

Migrate from `v0.19.2` to this version locally to verify the storage is not broken as a result of this migration
- Tested reading the orders and there was no issue

### CODE REVIEW

**Comment by brashidnia-figure** [comment]
Current dependencies on/for this PR:
* main
  * **PR #119** <a href="https://app.graphite.dev/github/pr/FigureTechnologies/ats-smart-contract/119" target="_blank">Graphite</a> 
    * **PR #120** <a href="https://app.graphite.dev/github/pr/FigureTechnologies/ats-smart-contract/120" target="_blank">Graphite</a>  👈
      * **PR #121** <a href="https://app.graphite.dev/github/pr/FigureTechnologies/ats-smart-contract/121" target="_blank">Graphite</a> 
        * **PR #122** <a href="https://app.graphite.dev/github/pr/FigureTechnologies/ats-smart-contract/122" target="_blank">Graphite</a> 

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/FigureTechnologies/ats-smart-contract/120?utm_source=stack-comment).

File: src/contract.rs [hunk 1]
```diff
@@ -1043,9 +1034,8 @@ fn modify_contract(
         return Err(ContractError::Unauthorized);
     }
 
-    let ask_storage = get_ask_storage_read(deps.storage);
-    let ask_orders: Vec<AskOrderV1> = ask_storage
-        .range(None, None, Order::Ascending)
+    let ask_orders: Vec<AskOrderV1> = ASKS_V1
+        .range(deps.storage, None, None, Order::Ascending)
```
Comment (brashidnia-figure) [inline src/contract.rs:1038]: Gas reduction is in https://github.com/FigureTechnologies/ats-smart-contract/pull/121/

File: src/ask_order.rs [hunk 2]
```diff
@@ -2,14 +2,17 @@ use crate::contract_info::require_version;
...
-use cosmwasm_storage::{bucket, bucket_read, Bucket, ReadonlyBucket};
+use cosmwasm_std::{Addr, Coin, DepsMut, Uint128};
+use cw_storage_plus::Map;
 use provwasm_std::ProvenanceQuery;
 use schemars::JsonSchema;
 use semver::Version;
 use serde::{Deserialize, Serialize};
 
 pub static NAMESPACE_ORDER_ASK: &[u8] = b"ask";
+pub const NAMESPACE_ORDER_ASK_V2: &str = "ask";
```
Comment (mwoods-figure) [inline src/ask_order.rs:13]: The mixing of `NAMESPACE_ORDER_ASK_V2`, which is used to create a storage map that's suffixed with `_V1` is somewhat confusing.  Also, is the existing usage of `NAMESPACE_ORDER_ASK` entirely removed? If so, you could just redefine `NAMESPACE_ORDER_ASK` as a `&str`.

**Review by mwoods-figure — DISMISSED** [review]
it's great that the migration to `cw_storage_plus` is pretty straightforward.

**Review by jtalis — DISMISSED** [review]

File: src/ask_order.rs (hunk 2 above)
Comment (brashidnia-figure) [reply src/ask_order.rs:13]: my mistake, i need to remove the old one

**Review by mwoods-figure — APPROVED** [review]

**Pull request merged**